File: hw/astudillo/U3/HW21Strategy/Strategy/src/main/java/ec/edu/espe/template/strategy/controller/SortingContext.java
package ec.edu.espe.template.strategy.controller;

import ec.edu.espe.template.strategy.model.SortingStrategy;
import ec.edu.espe.template.strategy.model.BubbleSort;
import ec.edu.espe.template.strategy.model.InsertionSort;
import ec.edu.espe.template.strategy.model.QuickSort;

/**
 *
 * @author Gerald Astudillo
 */

public class SortingContext {
    
    private SortingStrategy ss;
    
    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }
    
    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) {
            ss = new BubbleSort();
        }
        if (n >= 30 && n < 100) {
            ss = new InsertionSort();
        }
        if (n >= 100) {
            ss = new QuickSort();
        }
        return ss;
    }
}

File: ws/bonilla/U3/WS34Strategy/Strategyjs/model/SortingContext.js
const BubbleSort = require('../controller/BubbleSort');
const InsertionSort = require('../controller/InsertionSort');
const QuickSort = require('../controller/QuickSort');

class SortingContext {
    constructor() {
        this.ss = null;
    }

    sort(data) {
        const size = data.length;
        this.ss = this.setSortStrategy(size);
        return this.ss.sort(data);
    }

    setSortStrategy(n) {
        if (n > 0 && n < 30) {
            return new BubbleSort();
        }
        if (n >= 30 && n < 100) {
            return new InsertionSort();
        }
        if (n >= 100) {
            return new QuickSort();
        }
    }
}

module.exports = SortingContext;
